### PR TITLE
docs: replace `PSR*` with `PSR-*`

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- unify notation

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [] Conforms to style guide

### CODE REVIEW

File: user_guide_src/source/general/modules.rst [hunk 1]
```diff
@@ -129,7 +129,7 @@ Discovery and Composer
 ======================
 
 Packages that were installed via Composer will also be discovered by default. This only requires that the namespace
-that Composer knows about is a PSR4 namespace. PSR0 namespaces will not be detected.
+that Composer knows about is a PSR-4 namespace. PSR-0 namespaces will not be detected.
```
Comment (paulbalandan) [inline user_guide_src/source/general/modules.rst:132]: off-topic to this PR..

This notion on `PSR-0` is quite misleading. Composer can also detect PSR-0 namespaces via the `autoload.psr-0` key in composer.json. In fact, `mikey179/vfsstream` is using `PSR-0`. I think this should be removed in a separate PR.

**Review by paulbalandan — APPROVED** [review]

File: user_guide_src/source/general/modules.rst (hunk 1 above)
Comment (kenjis) [reply user_guide_src/source/general/modules.rst:132]: Okay, I will do.

Comment (kenjis) [reply user_guide_src/source/general/modules.rst:132]: Wait a second. This is not about Composer, but about `$discoverInComposer` in CI4.
We need to take a look.

Comment (kenjis) [reply user_guide_src/source/general/modules.rst:132]: It calls `$composer->getPrefixesPsr4()`, so PSR-0 namespaces will not be detected, right?
https://github.com/codeigniter4/CodeIgniter4/blob/7c08819c7e39770fb68cb3bc3aebb35a0aa32c44/system/Autoloader/Autoloader.php#L308

Comment (paulbalandan) [reply user_guide_src/source/general/modules.rst:132]: Hmm. Then we need a little bit of rewording.

`Packages installed via Composer using PSR-4 namespaces will also be discovered by default. Discovery of PSR-0 namespaced packages are not currently supported.`

Comment (kenjis) [reply user_guide_src/source/general/modules.rst:132]: PSR-0 is deprecated. We will not support it. I created a PR #5389.
